pylint/checkers/typecheck.py

```diff
@@ -523,6 +525,77 @@ def _emit_no_member(
     return True
 
 
+def _get_all_attribute_assignments(
+    node: astroid.FunctionDef, name: str | None = None
```

Comment:
Sorry I just noticed, we seem to be mixing `astroid` and `nodes` in this function. See L552 for example where we do `nodes.ClassDef`. Sorry for asking so late in the process, but could you get all nodes from `nodes.`? That's how we normally do this. I could also do this for you if you want to!

Reply:
nw, better to finish this off than starting the XML processing that's next in my queue 😛